Formula/w/whisperkit.rb

```diff
@@ -0,0 +1,18 @@
+class Whisperkit < Formula
+  desc "Swift native on-device speech recognition with Whisper for Apple Silicon"
+  homepage "https://github.com/argmaxinc/WhisperKit"
+  url "https://github.com/argmaxinc/WhisperKit.git", tag: "v0.3.3", revision: "bfa357e897dadae0dca6314424bb119149e375d2"
+  license "MIT"
+
+  depends_on xcode: ["15.0", :build]
+  depends_on arch: :arm64
+
+  def install
+    system "swift", "build", "-c", "release", "--product", "transcribe", "--disable-sandbox"
+    bin.install ".build/release/transcribe"
+  end
+
+  test do
+    system "#{bin}/transcribe", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
running tests would require to download a CoreML model which is at least ~66MB, I'm a bit worried about possible timeouts here

Reply:
ok, I've added some basic tests, let's see